File: icore/src/main/java/com/ddu/icore/ui/adapter/common/ViewHolder.java
package com.ddu.icore.ui.adapter.common;

import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.SparseArray;
import android.view.View;
import android.widget.ImageView;
import android.widget.RadioButton;
import android.widget.TextView;

import com.ddu.icore.util.sys.ViewUtils;

public class ViewHolder extends RecyclerView.ViewHolder {

    public View mConvertView;
    private SparseArray<View> mViews;

    public ViewHolder(@NonNull View itemView) {
        super(itemView);
        mConvertView = itemView;
        mViews = new SparseArray<>();
    }

    @NonNull
    public <T extends View> T getView(int viewId) {
        View view = mViews.get(viewId);
        if (null == view) {
            view = ViewUtils.findViewById(mConvertView, viewId);
            mViews.append(viewId, view);
        }
        return (T) view;
    }

    @NonNull
    public ViewHolder setText(int viewId, String text) {
        TextView tv = getView(viewId);
        tv.setText(text);
        return this;
    }

    @NonNull
    public ViewHolder setText(int viewId, int resId) {
        TextView tv = getView(viewId);
        tv.setText(resId);
        return this;
    }

    @NonNull
    public ViewHolder setText(int viewId, CharSequence charSequence) {
        TextView tv = getView(viewId);
        tv.setText(charSequence);
        return this;
    }

    @NonNull
    public ViewHolder setImageResource(int viewId, int resId) {
        ImageView view = getView(viewId);
        view.setImageResource(resId);
        return this;
    }

    @NonNull
    public ViewHolder setImageBitmap(int viewId, Bitmap bitmap) {
        ImageView view = getView(viewId);
        view.setImageBitmap(bitmap);
        return this;
    }

    @NonNull
    public ViewHolder setImageDrawable(int viewId, Drawable drawable) {
        ImageView view = getView(viewId);
        view.setImageDrawable(drawable);
        return this;
    }

    public ViewHolder setChecked(int viewId, boolean checked) {
        RadioButton radioButton = getView(viewId);
        radioButton.setChecked(checked);
        return this;
    }

    @NonNull
    public ViewHolder setBackgroud(int viewId, Drawable background) {
        View view = getView(viewId);
        view.setBackground(background);
        return this;
    }

    @NonNull
    public ViewHolder setBackgroud(int viewId, int color) {
        View view = getView(viewId);
        view.setBackgroundColor(color);
        return this;
    }

    @NonNull
    public ViewHolder setVisibility(int viewId, int visibility) {
        View view = getView(viewId);
        view.setVisibility(visibility);
        return this;
    }

    @NonNull
    public ViewHolder setOnClickListener(int viewId, View.OnClickListener listener) {
        View view = getView(viewId);
        view.setOnClickListener(listener);
        return this;
    }

    @NonNull
    public ViewHolder setOnClickListener(View.OnClickListener listener) {
        itemView.setOnClickListener(listener);
        return this;
    }

}

File: icore/src/main/java/com/ddu/icore/ui/adapter/common/ViewHolder.kt
package com.ddu.icore.ui.adapter.common

import android.graphics.Bitmap
import android.graphics.drawable.Drawable
import android.util.SparseArray
import android.view.View
import android.widget.ImageView
import android.widget.RadioButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.ddu.icore.common.find

class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    private val mViews: SparseArray<View> = SparseArray()

    fun <T : View> getView(viewId: Int): T? {
        var view = mViews.get(viewId)
        if (null == view) {
            view = itemView.find(viewId)
            mViews.append(viewId, view)
        }
        return view as? T
    }

    fun setText(viewId: Int, text: String): ViewHolder {
        val tv = getView<TextView>(viewId)
        tv?.text = text
        return this
    }

    fun setText(viewId: Int, resId: Int): ViewHolder {
        val tv = getView<TextView>(viewId)
        tv?.setText(resId)
        return this
    }

    fun setTextColor(viewId: Int, color: Int): ViewHolder {
        val tv = getView<TextView>(viewId)
        tv?.setTextColor(color)
        return this
    }

    fun setText(viewId: Int, charSequence: CharSequence): ViewHolder {
        val tv = getView<TextView>(viewId)
        tv?.text = charSequence
        return this
    }

    fun setImageResource(viewId: Int, resId: Int): ViewHolder {
        val view = getView<ImageView>(viewId)
        view?.setImageResource(resId)
        return this
    }

    fun setImageBitmap(viewId: Int, bitmap: Bitmap): ViewHolder {
        val view = getView<ImageView>(viewId)
        view?.setImageBitmap(bitmap)
        return this
    }

    fun setImageDrawable(viewId: Int, drawable: Drawable): ViewHolder {
        val view = getView<ImageView>(viewId)
        view?.setImageDrawable(drawable)
        return this
    }

    fun setChecked(viewId: Int, checked: Boolean): ViewHolder {
        val radioButton = getView<RadioButton>(viewId)
        radioButton?.isChecked = checked
        return this
    }

    fun setBackground(viewId: Int, background: Drawable): ViewHolder {
        val view = getView<View>(viewId)
        view?.background = background
        return this
    }

    fun setBackground(viewId: Int, color: Int): ViewHolder {
        val view = getView<View>(viewId)
        view?.setBackgroundColor(color)
        return this
    }

    fun setVisibility(viewId: Int, visibility: Int): ViewHolder {
        val view = getView<View>(viewId)
        view?.visibility = visibility
        return this
    }

    fun setOnClickListener(viewId: Int, listener: View.OnClickListener): ViewHolder {
        val view = getView<View>(viewId)
        view?.setOnClickListener(listener)
        return this
    }

    fun setOnClickListener(listener: View.OnClickListener): ViewHolder {
        itemView.setOnClickListener(listener)
        return this
    }

}
